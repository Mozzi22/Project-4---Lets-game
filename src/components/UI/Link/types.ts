import React from 'react';
import { colors } from 'src/components/UI/baseLayout';

export interface ILink {
    id: string;
    name?: string;
    onClick: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
    href?: string;
    title?: string;
    type?: 'button';
    value?: string;
    borderRadius?: string;
    isDisabled?: boolean;
    color?: string;
    fontSize?: string;
    fontWeight?: string;
    bgColorDisabled?: string;
    bgColor?: string;
    bgColorHover?: string;
    backgroundImage?: string;
    backgroundSize?: string;
    focusColorDefault?: string;
    height?: string;
    width?: string;
    padding?: string;
    margin?: string;
    content?: string;
    transition?: string;
    theme: 'dark' | 'light';
    colors: typeof colors;
}
