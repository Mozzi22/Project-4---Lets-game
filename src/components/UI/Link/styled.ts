import styled from 'styled-components';

import {
  hoverBgDefault,
} from '../baseLayout';
import { ILink } from './types';

export const StLink = styled.button < ILink > `
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Poppins', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ colors, theme }) => colors[theme].colorTextForm};
  border-radius: ${({ borderRadius = '0' }) => borderRadius};
  border: ${({ border = '0' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  background-color: ${({ bgColor = 'transparent' }) => bgColor};
  background-image: ${({ colors, theme }) => colors[theme].backgroundImage};
  background-size:  ${({ backgroundSize = 'cover' }) => backgroundSize};
  &:hover {
    border: ${({ border = 'none' }) => border};
    }
    background-color: ${({ bgColor = hoverBgDefault }) => bgColor};
  &:focus {
    color:  ${({ colors, theme }) => colors[theme].backgroundButton};
    border: ${({ border = 'none' }) => border};
    background-color: ${({ bgColor = 'transparent' }) => bgColor};
    }
`;
