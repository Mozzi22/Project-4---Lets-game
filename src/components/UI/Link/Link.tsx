import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/Hooks/useTheme';
import { StLink } from './styled';
import { ILink } from './types';

const Link = ({
    id,
    href,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    bgColorDisabled,
    bgColor,
    backgroundImage,
    backgroundSize,
    height,
    width,
    padding,
    margin,
    content,
    transition,
}: ILink) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();
    return (
        <StLink
            id={id}
            href={href}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            backgroundImage={backgroundImage}
            backgroundSize={backgroundSize}
            borderRadius={borderRadius}
            disabled={isDisabled}
            color={color}
            fontSize={fontSize}
            bgColor={bgColor}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            theme={theme}
            colors={colors}
        >
            {t(content)}
        </StLink>
    );
};

export default Link;
