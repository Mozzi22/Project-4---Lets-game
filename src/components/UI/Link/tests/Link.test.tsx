import React from 'react';
import { shallow } from 'enzyme';
import Link from '../Link';

describe('Link', () => {
    let props;
    beforeEach(() => {
        props = {
            id: 'id',
            onClick: jest.fn(),
            title: 'Link',
            name: 'name',
            type: 'Button',
            value: 'value',
            isDisabled: 'false',
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<Link {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StLink', () => {
        const component = shallow(<Link {...props}/>);
        expect(component.find('styled__StLink')).toHaveLength(1);
    });
    it('should call onClick', () => {
        const component = shallow(<Link {...props}/>);
        component.find('styled__StLink').simulate('click');
        expect(props.onClick).toHaveBeenCalled();
    });
});
