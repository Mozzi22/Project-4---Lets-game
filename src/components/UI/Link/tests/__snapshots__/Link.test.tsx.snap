// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should match snapshot 1`] = `"<button id=\\"id\\" type=\\"Button\\" value=\\"value\\" title=\\"Link\\" disabled=\\"\\" class=\\"styled__StLink-sc-dq9w1g-0 hIjsFD\\"></button>"`;
